### PR TITLE
Scene update resource finder

### PR DESCRIPTION
**Pull request by nkoenig**
# 🎉 New feature

## Summary

This adds a `/gazebo/resource_paths/resolve` service that will return the full path to a file on disk when provided with a URI. Only Fuel and IGN_GAZEBO_RESOURCE_PATH are checked.

This also stores the `<scene>` information the SceneBroadcaster so that the `scene/info` service can send over `<scene>` information.

## Test

Path resolver
```
ign service -s /gazebo/resource_paths/resolve --reqtype ignition.msgs.StringMsg --reptype ignition.msgs.StringMsg --timeout 2000 --req "data:'/tmp'"
```

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: test/integration/scene_broadcaster_system.cc [hunk 1]
```diff
@@ -914,6 +914,44 @@ TEST_P(SceneBroadcasterTest,
...
 }
 
+/////////////////////////////////////////////////
+TEST_P(SceneBroadcasterTest,
+    IGN_UTILS_TEST_DISABLED_ON_WIN32(SceneInfoHasSceneSdf))
+{
+  // Start server
+  ignition::gazebo::ServerConfig serverConfig;
+  serverConfig.SetSdfFile(std::string(PROJECT_SOURCE_PATH) +
+      "/test/worlds/conveyor.sdf");
```
Comment (ahcorde) [inline test/integration/scene_broadcaster_system.cc:924]: `jointPaths`

File: src/Server_TEST.cc [hunk 2]
```diff
@@ -1033,6 +1033,79 @@ TEST_P(ServerFixture, AddResourcePaths)
   }
 }
 
+/////////////////////////////////////////////////
+TEST_P(ServerFixture, ResolveResourcePaths)
+{
+  ignition::common::setenv("IGN_GAZEBO_RESOURCE_PATH",
+      "/tmp/some/path:/home/user/another_path");
```
Comment (ahcorde) [inline src/Server_TEST.cc:1040]: just asking, this will only work on Linux, what's about MacOS or WIndows ?

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: src/Server_TEST.cc (hunk 2 above)
Comment (nkoenig) [reply src/Server_TEST.cc]: Removed the Linux paths in [db49ac7](https://github.com/gazebosim/gz-sim/pull/1508/commits/db49ac7bddd625cec15414dcf51b24eaf31bf3db)

File: test/integration/scene_broadcaster_system.cc (hunk 1 above)
Comment (nkoenig) [reply test/integration/scene_broadcaster_system.cc]: Fixed in [db49ac7](https://github.com/gazebosim/gz-sim/pull/1508/commits/db49ac7bddd625cec15414dcf51b24eaf31bf3db)

**Review by caguero — COMMENTED** [review]
Added a minor fix in https://github.com/gazebosim/gz-sim/pull/1508/commits/99a868f4468fba957ca941248f6f2e7b94053bd7

**Review by caguero — APPROVED** [review]

**Pull request merged**